tests/unit/test_mypy.py

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2012-2022 Snowflake Computing Inc. All rights reserved.
+#
+import logging
+import subprocess
+
+def test_mypy_on_module():
+    """
+    tests that snowflake.snowpark module can be linted with mypy to detect possible type issues.
+    """
+    proc = subprocess.Popen(
+        ["mypy", "-p", "snowflake.snowpark", "--explicit-package-bases"],
+        stdout=subprocess.PIPE,
+        stdin=subprocess.PIPE,
+        encoding="utf-8",
+    )
+    try:
+        out, err = proc.communicate(timeout=60)
+    except subprocess.TimeoutExpired:
+        logging.error("mypy process timed out, consider increasing limit.")
+        proc.kill()
+        out, err = proc.communicate()
+
+    # for inspection, output mypy result
+    if out:
+        logging.info(out)
+    if err:
+        logging.error(err)
+
+    # mypy currently produces errors, therefore allow code 1.
```

Comment:
You can reproduce them via `mypy -p snowflake.snowpark --explicit-package-bases`, but it's a long list of typing issues

```
src/snowflake/snowpark/row.py:76: error: Signature of "__getitem__" incompatible with supertype "tuple"  [override]
src/snowflake/snowpark/row.py:76: note:      Superclass:
src/snowflake/snowpark/row.py:76: note:          @overload
src/snowflake/snowpark/row.py:76: note:          def __getitem__(self, SupportsIndex, /) -> Any
src/snowflake/snowpark/row.py:76: note:          @overload
src/snowflake/snowpark/row.py:76: note:          def __getitem__(self, slice, /) -> Tuple[Any, ...]
src/snowflake/snowpark/row.py:76: note:      Subclass:
src/snowflake/snowpark/row.py:76: note:          def __getitem__(self, Union[int, str, slice], /) -> Any
src/snowflake/snowpark/row.py:211: error: Incompatible return value type (got "Union[Dict[Any, Any], Iterable[Dict[Any, Any]]]", expected "Dict[Any, Any]")  [return-value]
src/snowflake/snowpark/row.py:224: error: List comprehension has incompatible type List[Union[Dict[Any, Any], Iterable[Dict[Any, Any]]]]; expected List[Dict[Any, Any]]  [misc]
src/snowflake/snowpark/row.py:243: error: Incompatible return value type (got "Optional[Any]", expected "bool")  [return-value]
src/snowflake/snowpark/_internal/code_generation.py:35: error: List or tuple literal expected as the defaults argument to namedtuple()  [misc]
src/snowflake/snowpark/_internal/code_generation.py:150: error: Argument 1 to "get_func_references" has incompatible type "object"; expected "FunctionType"  [arg-type]
src/snowflake/snowpark/_internal/code_generation.py:180: error: Invalid index type "Optional[int]" for "Tuple[str, ...]"; expected type "SupportsIndex"  [index]
src/snowflake/snowpark/_internal/code_generation.py:379: error: Argument 2 to "range" has incompatible type "Optional[int]"; expected "SupportsIndex"  [arg-type]
src/snowflake/snowpark/_internal/code_generation.py:383: error: Unsupported operand types for - ("None" and "int")  [operator]
src/snowflake/snowpark/_internal/code_generation.py:383: note: Left operand is of type "Optional[int]"
src/snowflake/snowpark/_internal/code_generation.py:439: error: Missing positional argument "alias" in call to "ImportNameAliasPair"  [call-arg]
src/snowflake/snowpark/_internal/code_generation.py:475: error: Missing positional argument "alias" in call to "ImportNameAliasPair"  [call-arg]
src/snowflake/snowpark/_internal/error_message.py:272: error: Argument 1 to "SnowparkSQLException" has incompatible type "Optional[str]"; expected "str"  [arg-type]
src/snowflake/snowpark/_internal/error_message.py:278: error: Argument 1 to "SnowparkSQLException" has incompatible type "Optional[str]"; expected "str"  [arg-type]
src/snowflake/snowpark/types.py:328: error: Name "pandas.Series" is not defined  [name-defined]
src/snowflake/snowpark/types.py:335: error: "TypeVarTuple" support is experimental, use --enable-incomplete-feature=TypeVarTuple to enable  [misc]
src/snowflake/snowpark/types.py:337: error: Name "pandas.DataFrame" is not defined  [name-defined]
src/snowflake/snowpark/types.py:337: error: Free type variable expected in Generic[...]  [misc]
src/snowflake/snowpark/context.py:19: error: Incompatible return value type (got "Optional[Session]", expected "Session")  [return-value]
src/snowflake/snowpark/_internal/utils.py:286: error: The return type of a generator function should be "Generator" or one of its supertypes  [misc]
src/snowflake/snowpark/_internal/utils.py:448: error: Incompatible return value type (got "Optional[Dict[str, str]]", expected "Dict[str, str]")  [return-value]
src/snowflake/snowpark/file_operation.py:192: error: Incompatible types in assignment (expression has type "str", target has type "int")  [assignment]
src/snowflake/snowpark/_internal/type_utils.py:211: error: Missing return statement  [return]
src/snowflake/snowpark/_internal/type_utils.py:240: error: Incompatible types in assignment (expression has type "Type[DataType]", target has type "Type[_FractionalType]")  [assignment]
src/snowflake/snowpark/_internal/type_utils.py:249: error: Incompatible types in assignment (expression has type "Type[DataType]", target has type "Type[_FractionalType]")  [assignment]
src/snowflake/snowpark/_internal/type_utils.py:254: error: Incompatible types in assignment (expression has type "Type[StringType]", target has type "Type[_FractionalType]")  [assignment]
src/snowflake/snowpark/_internal/type_utils.py:295: error: Incompatible types in assignment (expression has type "dict_items[Any, Any]", variable has type "zip[Tuple[Any, None]]")  [assignment]
src/snowflake/snowpark/_internal/type_utils.py:334: error: "DataType" has no attribute "fields"  [attr-defined]
src/snowflake/snowpark/_internal/type_utils.py:355: error: "DataType" has no attribute "element_type"  [attr-defined]
src/snowflake/snowpark/_internal/type_utils.py:361: error: "DataType" has no attribute "key_type"  [attr-defined]
src/snowflake/snowpark/_internal/type_utils.py:362: error: "DataType" has no attribute "value_type"  [attr-defined]
src/snowflake/snowpark/_internal/type_utils.py:477: error: Argument 1 to "parse_arg_annotation" has incompatible type "slice"; expected "expr"  [arg-type]
src/snowflake/snowpark/_internal/type_utils.py:543: error: Need type annotation for "DATA_TYPE_STRING_OBJECT_MAPPINGS" (hint: "DATA_TYPE_STRING_OBJECT_MAPPINGS: Dict[<type>, <type>] = ...")  [var-annotated]
src/snowflake/snowpark/_internal/type_utils.py:547: error: Incompatible types in assignment (expression has type "Type[ShortType]", target has type "Type[IntegerType]")  [assignment]
src/snowflake/snowpark/_internal/type_utils.py:548: error: Incompatible types in assignment (expression has type "Type[ByteType]", target has type "Type[IntegerType]")  [assignment]
src/snowflake/snowpark/_internal/type_utils.py:549: error: Incompatible types in assignment (expression has type "Type[LongType]", target has type "Type[IntegerType]")  [assignment]
src/snowflake/snowpark/_internal/type_utils.py:550: error: Incompatible types in assignment (expression has type "Type[DecimalType]", target has type "Type[IntegerType]")  [assignment]
src/snowflake/snowpark/_internal/type_utils.py:551: error: Incompatible types in assignment (expression has type "Type[DecimalType]", target has type "Type[IntegerType]")  [assignment]
src/snowflake/snowpark/_internal/type_utils.py:552: error: Incompatible types in assignment (expression has type "Type[MapType]", target has type "Type[IntegerType]")  [assignment]
src/snowflake/snowpark/_internal/type_utils.py:553: error: Incompatible types in assignment (expression has type "Type[ArrayType]", target has type "Type[IntegerType]")  [assignment]
src/snowflake/snowpark/_internal/type_utils.py:561: error: Missing return statement  [return]
src/snowflake/snowpark/_internal/type_utils.py:580: error: Argument 2 to NewType(...) must be subclassable (got "Union[Column, str]")  [valid-newtype]
src/snowflake/snowpark/_internal/type_utils.py:581: error: Argument 2 to NewType(...) must be subclassable (got "Union[Column, str]")  [valid-newtype]
src/snowflake/snowpark/_internal/type_utils.py:584: error: Argument 2 to NewType(...) must be subclassable (got "Union[Column, str]")  [valid-newtype]
src/snowflake/snowpark/_internal/type_utils.py:587: error: Argument 2 to NewType(...) must be subclassable (got VALID_PYTHON_TYPES_FOR_LITERAL_VALUE?)  [valid-newtype]
src/snowflake/snowpark/_internal/type_utils.py:587: error: Variable "snowflake.snowpark._internal.type_utils.VALID_PYTHON_TYPES_FOR_LITERAL_VALUE" is not valid as a type  [valid-type]
src/snowflake/snowpark/_internal/type_utils.py:587: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
src/snowflake/snowpark/_internal/type_utils.py:588: error: Argument 2 to NewType(...) must be subclassable (got "Union[Column, LiteralType]")  [valid-newtype]
src/snowflake/snowpark/_internal/telemetry.py:239: error: Incompatible types in assignment (expression has type "Optional[Any]", variable has type "TelemetryClient")  [assignment]
src/snowflake/snowpark/_internal/telemetry.py:308: error: Incompatible types in assignment (expression has type "List[str]", target has type "str")  [assignment]
src/snowflake/snowpark/_internal/telemetry.py:310: error: Incompatible types in assignment (expression has type "List[str]", target has type "str")  [assignment]
src/snowflake/snowpark/_internal/udf_utils.py:24: error: Skipping analyzing "cloudpickle": module is installed, but missing library stubs or py.typed marker  [import]
src/snowflake/snowpark/_internal/udf_utils.py:24: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
src/snowflake/snowpark/_internal/udf_utils.py:79: error: Argument 2 to "isinstance" has incompatible type "_SpecialForm"; expected "Union[type, Tuple[Union[type, Tuple[Any, ...]], ...]]"  [arg-type]
src/snowflake/snowpark/_internal/udf_utils.py:90: error: Value of type "Union[Callable[..., Any], Tuple[str, str]]" is not indexable  [index]
src/snowflake/snowpark/_internal/udf_utils.py:92: error: Value of type "Union[Callable[..., Any], Tuple[str, str]]" is not indexable  [index]
src/snowflake/snowpark/_internal/udf_utils.py:97: error: Value of type "Union[Callable[..., Any], Tuple[str, str]]" is not indexable  [index]
src/snowflake/snowpark/_internal/udf_utils.py:98: error: Value of type "Union[Callable[..., Any], Tuple[str, str]]" is not indexable  [index]
src/snowflake/snowpark/_internal/udf_utils.py:129: error: Incompatible return value type (got "Tuple[Optional[DataType], List[DataType]]", expected "Tuple[DataType, List[DataType]]")  [return-value]
src/snowflake/snowpark/_internal/udf_utils.py:199: error: Value of type "Optional[List[DataType]]" is not indexable  [index]
src/snowflake/snowpark/_internal/udf_utils.py:200: error: Argument 1 to "len" has incompatible type "Optional[List[DataType]]"; expected "Sized"  [arg-type]
src/snowflake/snowpark/_internal/udf_utils.py:201: error: Value of type "Optional[List[DataType]]" is not indexable  [index]
src/snowflake/snowpark/_internal/udf_utils.py:205: error: List comprehension has incompatible type List[Union[DataType, None, Any]]; expected List[DataType]  [misc]
src/snowflake/snowpark/_internal/udf_utils.py:206: error: Item "None" of "Optional[Iterable[DataType]]" has no attribute "__iter__" (not iterable)  [union-attr]
src/snowflake/snowpark/_internal/udf_utils.py:209: error: Incompatible return value type (got "Tuple[bool, bool, Optional[DataType], List[DataType]]", expected "Tuple[bool, bool, DataType, List[DataType]]")  [return-value]
src/snowflake/snowpark/_internal/udf_utils.py:213: error: Incompatible return value type (got "Tuple[bool, bool, Optional[DataType], List[DataType]]", expected "Tuple[bool, bool, DataType, List[DataType]]")  [return-value]
src/snowflake/snowpark/_internal/udf_utils.py:217: error: Incompatible return value type (got "Tuple[bool, bool, Optional[DataType], List[DataType]]", expected "Tuple[bool, bool, DataType, List[DataType]]")  [return-value]
src/snowflake/snowpark/_internal/udf_utils.py:232: error: Argument 2 to "isinstance" has incompatible type "_SpecialForm"; expected "Union[type, Tuple[Union[type, Tuple[Any, ...]], ...]]"  [arg-type]
src/snowflake/snowpark/_internal/udf_utils.py:237: error: Item "Tuple[str, ...]" of "Union[Callable[..., Any], Tuple[str, str]]" has no attribute "__code__"  [union-attr]
src/snowflake/snowpark/_internal/udf_utils.py:239: error: Item "Tuple[str, ...]" of "Union[Callable[..., Any], Tuple[str, str]]" has no attribute "__code__"  [union-attr]
src/snowflake/snowpark/_internal/udf_utils.py:250: error: Incompatible return value type (got "Tuple[bool, bool, Optional[DataType], List[DataType]]", expected "Tuple[bool, bool, DataType, List[DataType]]")  [return-value]
src/snowflake/snowpark/_internal/udf_utils.py:254: error: Incompatible return value type (got "Tuple[bool, bool, Optional[DataType], List[DataType]]", expected "Tuple[bool, bool, DataType, List[DataType]]")  [return-value]
src/snowflake/snowpark/_internal/udf_utils.py:259: error: List comprehension has incompatible type List[Optional[DataType]]; expected List[DataType]  [misc]
src/snowflake/snowpark/_internal/udf_utils.py:264: error: Incompatible return value type (got "Tuple[bool, bool, Optional[DataType], List[DataType]]", expected "Tuple[bool, bool, DataType, List[DataType]]")  [return-value]
src/snowflake/snowpark/_internal/udf_utils.py:268: error: "DataType" has no attribute "element_type"  [attr-defined]
src/snowflake/snowpark/_internal/udf_utils.py:385: error: Argument 1 to "generate_source_code" has incompatible type "Callable[..., Any]"; expected "Union[FunctionType, BuiltinFunctionType]"  [arg-type]
src/snowflake/snowpark/_internal/udf_utils.py:554: error: Argument 2 to "isinstance" has incompatible type "_SpecialForm"; expected "Union[type, Tuple[Union[type, Tuple[Any, ...]], ...]]"  [arg-type]
src/snowflake/snowpark/_internal/udf_utils.py:560: error: Argument 1 to "generate_python_code" has incompatible type "Union[Callable[..., Any], Tuple[str, str]]"; expected "Callable[..., Any]"  [arg-type]
src/snowflake/snowpark/_internal/analyzer/expression.py:27: error: Need type annotation for "COLUMN_DEPENDENCY_EMPTY"  [var-annotated]
src/snowflake/snowpark/_internal/analyzer/expression.py:31: error: Need type annotation for "result" (hint: "result: Set[<type>] = ...")  [var-annotated]
src/snowflake/snowpark/_internal/analyzer/expression.py:36: error: Incompatible return value type (got "FrozenSet[str]", expected "Optional[Set[str]]")  [return-value]
src/snowflake/snowpark/_internal/analyzer/expression.py:39: error: Argument 1 to "update" of "set" has incompatible type "Optional[Set[str]]"; expected "Iterable[Any]"  [arg-type]
src/snowflake/snowpark/_internal/analyzer/expression.py:60: error: Incompatible return value type (got "FrozenSet[Any]", expected "Optional[Set[str]]")  [return-value]
src/snowflake/snowpark/_internal/analyzer/expression.py:84: error: Incompatible types in assignment (expression has type "None", variable has type "str")  [assignment]
src/snowflake/snowpark/_internal/analyzer/expression.py:97: error: Incompatible return value type (got "FrozenSet[str]", expected "Optional[Set[str]]")  [return-value]
src/snowflake/snowpark/_internal/analyzer/expression.py:169: error: Incompatible types in assignment (expression has type "Optional[Set[str]]", variable has type "FrozenSet[str]")  [assignment]
src/snowflake/snowpark/_internal/analyzer/expression.py:186: error: Incompatible return value type (got "FrozenSet[str]", expected "Optional[Set[str]]")  [return-value]
src/snowflake/snowpark/_internal/analyzer/expression.py:194: error: Argument 2 to "isinstance" has incompatible type "Tuple[object, ...]"; expected "Union[type, Tuple[Union[type, Tuple[Any, ...]], ...]]"  [arg-type]
src/snowflake/snowpark/_internal/analyzer/expression.py:196: error: Argument 1 to "PLAN_CANNOT_CREATE_LITERAL" of "SnowparkClientExceptionMessages" has incompatible type "Type[Any]"; expected "str"  [arg-type]
src/snowflake/snowpark/_internal/analyzer/expression.py:283: error: Item "None" of "Optional[List[Expression]]" has no attribute "__iter__" (not iterable)  [union-attr]
src/snowflake/snowpark/_internal/analyzer/expression.py:287: error: List or tuple expected as variadic arguments  [misc]
src/snowflake/snowpark/_internal/analyzer/expression.py:312: error: Need type annotation for "exps" (hint: "exps: List[<type>] = ...")  [var-annotated]
src/snowflake/snowpark/_internal/analyzer/expression.py:337: error: List or tuple expected as variadic arguments  [misc]
src/snowflake/snowpark/_internal/analyzer/datatype_mapper.py:129: error: Argument 1 to "hexlify" has incompatible type "Union[List[Any], bytes, bytearray]"; expected "Union[bytes, Union[bytearray, memoryview, array[Any], mmap, _CData, PickleBuffer]]"  [arg-type]
src/snowflake/snowpark/stored_procedure.py:67: error: Function "builtins.any" is not valid as a type  [valid-type]
src/snowflake/snowpark/stored_procedure.py:67: note: Perhaps you meant "typing.Any" instead of "any"?
src/snowflake/snowpark/stored_procedure.py:84: error: Item "None" of "Optional[Session]" has no attribute "_conn"  [union-attr]
src/snowflake/snowpark/stored_procedure.py:422: error: Argument 2 to "_do_register_sp" of "StoredProcedureRegistration" has incompatible type "Optional[DataType]"; expected "DataType"  [arg-type]
src/snowflake/snowpark/stored_procedure.py:423: error: Argument 3 to "_do_register_sp" of "StoredProcedureRegistration" has incompatible type "Optional[List[DataType]]"; expected "List[DataType]"  [arg-type]
src/snowflake/snowpark/stored_procedure.py:424: error: Argument 4 to "_do_register_sp" of "StoredProcedureRegistration" has incompatible type "Union[str, Iterable[str], None]"; expected "str"  [arg-type]
src/snowflake/snowpark/stored_procedure.py:530: error: Argument 2 to "_do_register_sp" of "StoredProcedureRegistration" has incompatible type "Optional[DataType]"; expected "DataType"  [arg-type]
src/snowflake/snowpark/stored_procedure.py:531: error: Argument 3 to "_do_register_sp" of "StoredProcedureRegistration" has incompatible type "Optional[List[DataType]]"; expected "List[DataType]"  [arg-type]
src/snowflake/snowpark/stored_procedure.py:532: error: Argument 4 to "_do_register_sp" of "StoredProcedureRegistration" has incompatible type "Union[str, Iterable[str], None]"; expected "str"  [arg-type]
src/snowflake/snowpark/stored_procedure.py:637: error: Argument 3 to "cleanup_failed_permanent_registration" has incompatible type "Optional[str]"; expected "str"  [arg-type]
src/snowflake/snowpark/stored_procedure.py:641: error: Argument 1 to "StoredProcedure" has incompatible type "Union[Callable[..., Any], Tuple[str, str]]"; expected "Callable[..., Any]"  [arg-type]
src/snowflake/snowpark/_internal/analyzer/unary_expression.py:33: error: Argument 1 to "derive_dependent_columns" has incompatible type "Optional[Expression]"; expected "Expression"  [arg-type]
src/snowflake/snowpark/_internal/analyzer/sort_expression.py:31: error: Incompatible types in assignment (expression has type "Type[NullsFirst]", base class "SortDirection" defined the type as "NullOrdering")  [assignment]
src/snowflake/snowpark/_internal/analyzer/sort_expression.py:36: error: Incompatible types in assignment (expression has type "Type[NullsLast]", base class "SortDirection" defined the type as "NullOrdering")  [assignment]
src/snowflake/snowpark/_internal/analyzer/sort_expression.py:55: error: Argument 1 to "derive_dependent_columns" has incompatible type "Optional[Expression]"; expected "Expression"  [arg-type]
src/snowflake/snowpark/_internal/analyzer/snowflake_plan_node.py:16: error: Need type annotation for "children" (hint: "children: List[<type>] = ...")  [var-annotated]
src/snowflake/snowpark/_internal/analyzer/binary_plan_node.py:84: error: Cannot override writeable attribute with read-only property  [override]
src/snowflake/snowpark/_internal/analyzer/binary_plan_node.py:176: error: Cannot override writeable attribute with read-only property  [override]
src/snowflake/snowpark/window.py:32: error: Incompatible types in assignment (expression has type "UnboundedPreceding", variable has type "CurrentRow")  [assignment]
src/snowflake/snowpark/window.py:34: error: Incompatible types in assignment (expression has type "Literal", variable has type "CurrentRow")  [assignment]
src/snowflake/snowpark/window.py:39: error: Incompatible types in assignment (expression has type "UnboundedFollowing", variable has type "CurrentRow")  [assignment]
src/snowflake/snowpark/window.py:41: error: Incompatible types in assignment (expression has type "Literal", variable has type "CurrentRow")  [assignment]
src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py:410: error: Incompatible types in assignment (expression has type "float", variable has type "int")  [assignment]
src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py:445: error: Argument 2 to "to_sql" has incompatible type "Optional[DataType]"; expected "DataType"  [arg-type]
src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py:647: error: Argument 1 to "convert_sp_to_sf_type" has incompatible type "Optional[DataType]"; expected "DataType"  [arg-type]
src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py:1149: error: Argument 1 to "convert_sp_to_sf_type" has incompatible type "Optional[DataType]"; expected "DataType"  [arg-type]
src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py:1156: error: Argument 1 to "schema_expression" has incompatible type "Optional[DataType]"; expected "DataType"  [arg-type]
src/snowflake/snowpark/column.py:81: error: Incompatible return value type (got "Union[ColumnOrLiteral, DataFrame]", expected "Column")  [return-value]
src/snowflake/snowpark/column.py:82: error: Argument 2 to "isinstance" has incompatible type "Tuple[object, ...]"; expected "Union[type, Tuple[Union[type, Tuple[Any, ...]], ...]]"  [arg-type]
src/snowflake/snowpark/column.py:218: error: Incompatible types in assignment (expression has type "UnresolvedAttribute", variable has type "Star")  [assignment]
src/snowflake/snowpark/column.py:220: error: Incompatible types in assignment (expression has type "Expression", variable has type "Star")  [assignment]
src/snowflake/snowpark/column.py:233: error: Argument 1 of "__eq__" is incompatible with supertype "object"; supertype defines the argument type as "object"  [override]
src/snowflake/snowpark/column.py:233: note: This violates the Liskov substitution principle
src/snowflake/snowpark/column.py:233: note: (Skipping most remaining errors due to unresolved imports or missing stubs; fix these first)
src/snowflake/snowpark/async_job.py:289: error: Name "pandas.DataFrame" is not defined  [name-defined]
src/snowflake/snowpark/_internal/server_connection.py:399: error: Name "pandas.DataFrame" is not defined  [name-defined]
src/snowflake/snowpark/_internal/server_connection.py:428: error: Name "pandas.DataFrame" is not defined  [name-defined]
src/snowflake/snowpark/_internal/server_connection.py:574: error: Name "pandas.DataFrame" is not defined  [name-defined]
src/snowflake/snowpark/_internal/server_connection.py:575: error: Name "pandas.DataFrame" is not defined  [name-defined]
src/snowflake/snowpark/session.py:17: error: Skipping analyzing "cloudpickle": module is installed, but missing library stubs or py.typed marker  [import]
src/snowflake/snowpark/session.py:1167: error: Name "pandas.DataFrame" is not defined  [name-defined]
src/snowflake/snowpark/session.py:1313: error: Name "pandas.DataFrame" is not defined  [name-defined]
src/snowflake/snowpark/dataframe_reader.py:234: error: "dict" is not subscriptable, use "typing.Dict" instead  [misc]
src/snowflake/snowpark/dataframe.py:145: error: Cannot find implementation or library stub for module named "table"  [import]
src/snowflake/snowpark/dataframe.py:667: error: Name "to_pandas" already defined on line 657  [no-redef]
src/snowflake/snowpark/dataframe.py:735: error: Name "to_pandas_batches" already defined on line 725  [no-redef]
Found 134 errors in 25 files (checked 51 source files)
```

Reply:
I see. I guess fixing the typing issue is going to be a separate PR?

Reply:
Yea let's create a jira for that?